code/game/objects/items/weapons/tools/weldingtool.dm

```diff
@@ -87,37 +87,45 @@
 
 /obj/item/weldingtool/attackby(obj/item/W, mob/user)
 	if(welding)
-		to_chat(user, SPAN_DANGER("Stop welding first!"))
+		to_chat(user, SPAN_WARNING("Stop welding first!"))
 		return
 
 	if(isScrewdriver(W))
+		if(isrobot(loc))
+			to_chat(user, SPAN_WARNING("You cannot modify your own welder!"))
+			return
+
 		status = !status
+		
 		if(status)
 			to_chat(user, SPAN_NOTICE("You secure the welder."))
 		else
 			to_chat(user, SPAN_NOTICE("The welder can now be attached and modified."))
-		src.add_fingerprint(user)
+
+		add_fingerprint(user)
 		return
 
-	if((!status) && (istype(W,/obj/item/stack/material/rods)))
+	if(!status && istype(W, /obj/item/stack/material/rods))
 		var/obj/item/stack/material/rods/R = W
 		R.use(1)
-		var/obj/item/flamethrower/F = new/obj/item/flamethrower(user.loc)
-		user.drop_from_inventory(src, F)
-		F.weldtool = src
-		master = F
 		add_fingerprint(user)
+		user.drop_from_inventory(src)
```

Comment:
You've removed the second argument from this, src is meant to end up inside F.

Reply:
oopss

Reply:
Nothing that this is moved in flamethrower Initialize().